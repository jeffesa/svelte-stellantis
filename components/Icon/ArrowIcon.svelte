<script>
	export let name
	export let type = 'left'
	export let scale = 1.5
</script>

<i
	class:left="{type === 'left'}"
	class:right="{type === 'right'}"
	style="--scale:{scale}"
	rel="icon"
	role="{name}"
></i>

<style>
	$size: 1.5px;
	i {
		color: var(--secondary-default);
		box-sizing: border-box;
		position: relative;
		display: block;
		transform: scale(var(--scale, 1));
		width: 22px;
		height: @width;
		&:hover {
			color: var(--secondary-default);
		}
		&::after,
		&::before {
			content: '';
			display: block;
			box-sizing: border-box;
			position: absolute;
		}
		&.left::after,
		&.left::before {
			left: 3px;
		}
		&.right::after,
		&.right::before {
			right: 3px;
		}
		&::after {
			width: 10px;
			height: @width;
			bottom: 6px;
		}
		&.left::after {
			border-bottom: $size solid;
			border-left: @border-bottom;
			transform: rotate(45deg);
		}
		&.right::after {
			border-top: $size solid;
			border-right: @border-top;
			transform: rotate(45deg);
		}
		&::before {
			width: 15px;
			height: 2px;
			bottom: 10px;
			background: currentColor;
		}
	}
</style>
