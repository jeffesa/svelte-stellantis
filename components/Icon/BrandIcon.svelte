<script>
	export let name = ''
	export let colored = false
	export let sizes = ['22.59px']
	export let color = '#fff'
</script>

<style>
	i {
		@apply flex;
	}
	span {
		height: var(--sizeHeight);
		width: calc(var(--sizeWidth) * 0.2571);
		background: var(--color, '#fff');
		margin-right: @width;
		transform: skewX(-21deg);
	}
	span:last-child {
		margin-right: 0;
	}
</style>

{#if name === 'fiat'}
	<i class="{$$props.class || ''}" rel="icon">
		<span
			style="
        --color: {colored ? 'var(--fiatRed)' : color}; 
        --sizeWidth: {sizes[0]}; 
        --sizeHeight: {sizes.length > 1 ? sizes[1] : sizes[0]}
      "
		></span>
		<span
			style="
        --color: {colored ? '#fff' : color}; 
        --sizeWidth: {sizes[0]}; 
        --sizeHeight: {sizes.length > 1 ? sizes[1] : sizes[0]}
      "
		></span>
		<span
			style="
        --color: {colored ? '#fff' : color}; 
        --sizeWidth: {sizes[0]}; 
        --sizeHeight: {sizes.length > 1 ? sizes[1] : sizes[0]}
      "
		></span>
		<span
			style="
        --color: {colored ? 'var(--fiatGreen)' : color}; 
        --sizeWidth: {sizes[0]}; 
        --sizeHeight: {sizes.length > 1 ? sizes[1] : sizes[0]}
      "
		></span>
	</i>
{:else if name === 'citroen'}
<!-- <i class="{$$props.class || ''}" rel="icon">
	<svg id="Camada_1" data-name="Camada 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 68.51 48.74"><path d="M44.51,42.21c-1.68,0-3.39,0-3.41,0a3.49,3.49,0,0,0-3.72,3.19,3.52,3.52,0,0,0,3.72,3.26s1.73,0,3.41,0a3.54,3.54,0,0,0,3.75-3.26C48.26,43.67,46.62,42.25,44.51,42.21Zm-1.15,5.56c-.13,0-.62,0-1.08,0a2.42,2.42,0,0,1-2.46-2.35,2.37,2.37,0,0,1,2.46-2.3s.59,0,1.08,0a2.36,2.36,0,0,1,2.46,2.3A2.39,2.39,0,0,1,43.36,47.77Z" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M66.41,48.57h2.1V45.49a3.38,3.38,0,0,0-3.63-3.19h-5.3c-.09,0-.29,0-.29.24v5.81a.23.23,0,0,0,.22.24h2.22V43.74a.31.31,0,0,1,.31-.33h2.44a1.82,1.82,0,0,1,1.68,2v3a.24.24,0,0,0,.25.22" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M22.16,43.54h3.28v-1a.25.25,0,0,0-.25-.24H16.06a.24.24,0,0,0-.24.24v1H18a1.4,1.4,0,0,1,1.38,1.37v3.44a.24.24,0,0,0,.24.24h2.22V43.87a.33.33,0,0,1,.33-.33" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M12.05,48.35a.27.27,0,0,0,.25.24h.13l0,0a7.58,7.58,0,0,1,1,0h1.08V42.54a.24.24,0,0,0-.24-.24h-2a.25.25,0,0,0-.25.24v5.81Z" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M34.61,46.38l0,0v-.09h0a2.12,2.12,0,0,0,1.77-1.95,2.27,2.27,0,0,0-2.48-2H28a1.26,1.26,0,0,0-1.27,1.28v4.77a.23.23,0,0,0,.25.24H29.2V46.38h2.09A2.48,2.48,0,0,1,34,48.57h2.43A2.61,2.61,0,0,0,34.61,46.38ZM33,45.25H29.18V44a.57.57,0,0,1,.6-.57H33a.92.92,0,0,1,.91.9A.94.94,0,0,1,33,45.25Z" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M8.35,44.4h2.4a3.84,3.84,0,0,0-3.59-2.23l-.73,0a26.33,26.33,0,0,0-2.69,0,4,4,0,0,0-2.63.95A3.06,3.06,0,0,0,0,45.42,3.52,3.52,0,0,0,3.7,48.7c1.44,0,1.6,0,3.48,0a3.58,3.58,0,0,0,3.54-2.24H8.35a.26.26,0,0,0-.2.11A2.45,2.45,0,0,1,6,47.79c-.48,0-1.08,0-1.08,0a2.38,2.38,0,0,1-2.46-2.35v0A2.37,2.37,0,0,1,4.92,43.1c.13,0,.62,0,1.06,0a2.49,2.49,0,0,1,2.15,1.19.29.29,0,0,0,.22.11" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M58,47.44H52.2a.3.3,0,0,1-.33-.31v-1.2h5V44.82h-5V43.74a.34.34,0,0,1,.09-.22.37.37,0,0,1,.29-.11H58v-.89a.28.28,0,0,0-.06-.18.41.41,0,0,0-.22-.06h-8a.23.23,0,0,0-.24.24v4.74a1.29,1.29,0,0,0,1.29,1.29h7a.24.24,0,0,0,.16-.07.24.24,0,0,0,.06-.15s0-.8,0-.89" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M56.17,40a1.45,1.45,0,0,0-.84-.22,1.39,1.39,0,0,0-.84.22c-.23.13-.4.33-.67.33s-.42-.2-.66-.33a1.76,1.76,0,0,0-.85-.22,1.41,1.41,0,0,0-.84.22.58.58,0,0,0-.35.51.64.64,0,0,0,.35.53,1.47,1.47,0,0,0,.84.22,1.42,1.42,0,0,0,.82-.2c.23-.13.45-.33.69-.33a1.41,1.41,0,0,1,.67.33,1.54,1.54,0,0,0,.84.22,1.51,1.51,0,0,0,.86-.22.63.63,0,0,0,.36-.53.68.68,0,0,0-.38-.53" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M52.93,29.25c-.55-.62-13.87-15.31-18.67-15.31S16.15,28.61,15.58,29.25a5.67,5.67,0,0,0-1.35,3.28c0,1.35.08,2.7.08,2.72s0,.65.36.8a.42.42,0,0,0,.15,0,.69.69,0,0,0,.4-.15c.14-.13,13.7-12.1,19-12.1s18.87,12,19,12.08a.67.67,0,0,0,.4.15.33.33,0,0,0,.15,0c.31-.16.35-.73.35-.8s.09-1.37.09-2.7A5.36,5.36,0,0,0,52.93,29.25Zm.14,4.92c-3.5-3-13.9-11.53-18.81-11.53S18.94,31.18,15.44,34.17c0-.71,0-1.13,0-1.78a3,3,0,0,1,.84-2.08A62,62,0,0,1,32,20c2.37-1,2.24-.82,2.24-4.83h.17c3.35.31,12.32,9,17.64,14.92a4.5,4.5,0,0,1,1,2.48C53.11,33.1,53.09,33.7,53.07,34.17Z" style="fill:#6e6e6e;fill-rule:evenodd"/><path d="M52.91,15.29C52.36,14.67,39.06,0,34.26,0S16.17,14.67,15.6,15.29a5.7,5.7,0,0,0-1.35,3.28c0,1.35.09,2.7.09,2.7s0,.64.35.8a.42.42,0,0,0,.16,0,.66.66,0,0,0,.39-.15c.16-.13,13.68-12.08,19-12.08S53.11,21.83,53.27,22a.67.67,0,0,0,.4.15.33.33,0,0,0,.15,0c.31-.16.35-.73.35-.8s.09-1.37.09-2.7A5.7,5.7,0,0,0,52.91,15.29ZM53,20.21c-3.5-3-13.87-11.52-18.78-11.52S19,17.19,15.47,20.21c0-.71-.05-1.13-.05-1.77a3,3,0,0,1,.84-2.09A62.05,62.05,0,0,1,32,6c2.37-1,2.24-.82,2.24-4.83h.17c3.35.31,12.3,9,17.62,14.89a4.5,4.5,0,0,1,1,2.48C53.09,19.14,53.07,19.74,53,20.21Z" style="fill:#6e6e6e;fill-rule:evenodd"/></svg>
</i> -->
{:else if name === 'jeep'}
	<i class="{$$props.class || ''}" rel="icon">
		<svg xmlns="http://www.w3.org/2000/svg" width="28.157" height="31" viewBox="0 0 28.157 31">
			<g id="Grupo_2372" data-name="Grupo 2372" transform="translate(1964.33 -348.296)">
				<path
					id="Caminho_1313"
					data-name="Caminho 1313"
					d="M-1783.423,526.634s-3.132-.734-10.483-.734-10.483.734-10.483.734v-1.366s3.132-.734,10.483-.734,10.483.734,10.483.734Z"
					transform="translate(-156.381 -172.316)"
					fill="#fff"
				></path>
				<path
					id="Caminho_1314"
					data-name="Caminho 1314"
					d="M-1936.263,349.5a82.492,82.492,0,0,0-13.989-1.207,82.492,82.492,0,0,0-13.989,1.207c-.071,1.909-.152,4.859-.016,7.363.3,5.582,1.534,9.915,3.872,13.635a23.642,23.642,0,0,0,10.133,8.794,23.642,23.642,0,0,0,10.133-8.794c2.338-3.721,3.568-8.054,3.872-13.635C-1936.111,354.363-1936.192,351.412-1936.263,349.5Zm-1,7.866c-.281,5.177-1.422,9.2-3.591,12.647a21.928,21.928,0,0,1-9.4,8.156,21.809,21.809,0,0,1-9.354-8.156,25.525,25.525,0,0,1-3.636-12.647c-.126-2.322-.051-5.059.015-6.829a76.512,76.512,0,0,1,12.974-1.12,76.511,76.511,0,0,1,12.974,1.12C-1937.212,352.31-1937.137,355.047-1937.263,357.369Z"
					fill="#fff"
				></path>
			</g>
		</svg>
	</i>
	{:else if name === 'peugeot'}
	<i class="{$$props.class || ''}" rel="icon">
		<svg xmlns="http://www.w3.org/2000/svg" width="28.157" height="31" viewBox="0 0 28.157 31"><g id="Grupo_2372" data-name="Grupo 2372" transform="translate(1964.33 -348.296)"><path id="Caminho_1313" data-name="Caminho 1313" d="M-1783.423,526.634s-3.132-.734-10.483-.734-10.483.734-10.483.734v-1.366s3.132-.734,10.483-.734,10.483.734,10.483.734Z" transform="translate(-156.381 -172.316)" fill="#fff"></path><path id="Caminho_1314" data-name="Caminho 1314" d="M-1936.263,349.5a82.492,82.492,0,0,0-13.989-1.207,82.492,82.492,0,0,0-13.989,1.207c-.071,1.909-.152,4.859-.016,7.363.3,5.582,1.534,9.915,3.872,13.635a23.642,23.642,0,0,0,10.133,8.794,23.642,23.642,0,0,0,10.133-8.794c2.338-3.721,3.568-8.054,3.872-13.635C-1936.111,354.363-1936.192,351.412-1936.263,349.5Zm-1,7.866c-.281,5.177-1.422,9.2-3.591,12.647a21.928,21.928,0,0,1-9.4,8.156,21.809,21.809,0,0,1-9.354-8.156,25.525,25.525,0,0,1-3.636-12.647c-.126-2.322-.051-5.059.015-6.829a76.512,76.512,0,0,1,12.974-1.12,76.511,76.511,0,0,1,12.974,1.12C-1937.212,352.31-1937.137,355.047-1937.263,357.369Z" fill="#fff"></path></g></svg>
	</i>
{/if}
