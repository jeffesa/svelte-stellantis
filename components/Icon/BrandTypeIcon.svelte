<script>
	export let scale = 1
  export let color = '#fb7b3c'
</script>

<i rel="icon">
	<svg
    style="--scale:{scale}; --color: {color}"
		xmlns="http://www.w3.org/2000/svg"
		id="Group_909"
		width="81"
		height="56.59"
		viewBox="0 0 81 56.59"
	>
		<defs>
			<style>
				.cls-1 {
          fill: var(--color, '#fb7b3c');
				}
			</style>
		</defs>
		<path
			id="Path_146"
			d="M696.233 300h-16.8v56.555h7.666v-25.547h8.453c.753 4.67.821 24.037.821 25.548h7.769c.239-14.353.1-41.1-7.906-56.555m-9.24 24.449v-17.994h4.104a48.73 48.73 0 0 1 3.114 11.16c.65 3.709.993 6.833.993 6.833z"
			class="cls-1"
			transform="translate(-643.61 -299.999)"
		></path>
		<path
			id="Rectangle_55"
			d="M0 0H8.419V56.59H0z"
			class="cls-1"
			transform="translate(21.331 .001)"
		></path>
		<path
			id="Path_147"
			d="M299.581 300v56.59H308v-25.616h6.16v-6.559H308v-17.89h7.9V300z"
			class="cls-1"
			transform="translate(-299.581 -300.001)"
		></path>
		<path
			id="Path_148"
			d="M941.894 300v6.524h6.023l.034 50.031h8.3v-50.03h6.074V300z"
			class="cls-1"
			transform="translate(-881.324 -300.001)"
		></path>
	</svg>
</i>

<style>
	svg {
		transform: scale(var(--scale, 1));
	}
</style>
