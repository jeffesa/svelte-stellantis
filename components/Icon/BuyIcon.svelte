<script>
	export let scale = 1
</script>

<i rel="icon">
	<svg
		style="--scale:{scale}"
		xmlns="http://www.w3.org/2000/svg"
		width="19.993"
		height="19.993"
		viewBox="0 0 19.993 19.993"
	>
		<g
			id="Group_1082"
			data-name="Group 1082"
			transform="translate(-306.971 -385.092)"
		>
			<path
				id="Union_33"
				data-name="Union 33"
				d="M8054.478,10965.421a9.5,9.5,0,1,1,9.822,9.158c-.114,0-.227.006-.34.006A9.5,9.5,0,0,1,8054.478,10965.421Zm9.274-6.533a6.2,6.2,0,1,0,.213,0C8063.894,10958.884,8063.823,10958.885,8063.751,10958.888Zm-3.594,9.883,2.25-1.234a.277.277,0,0,1,.372.107.38.38,0,0,1,.033.107l.234,2.564A5.3,5.3,0,0,1,8060.157,10968.771Zm5.153-1.1a.274.274,0,0,1,.275-.27.329.329,0,0,1,.116.027l2.338,1.074a5.314,5.314,0,0,1-2.791,1.738Zm-6.644-2.394a5.271,5.271,0,0,1,.2-1.646l2.2,1.333a.276.276,0,0,1,.091.377.289.289,0,0,1-.074.08l-2.1,1.487A5.235,5.235,0,0,1,8058.667,10965.274Zm8.2-.061a.272.272,0,0,1-.091-.374.266.266,0,0,1,.075-.08l2.1-1.487a5.32,5.32,0,0,1,.319,1.631h-.005a5.3,5.3,0,0,1-.2,1.649Zm-3.862-.091a.965.965,0,1,1,1,.929h-.036A.962.962,0,0,1,8063,10965.123Zm-.655-2.344a.279.279,0,0,1-.107-.024l-2.333-1.071a5.265,5.265,0,0,1,2.773-1.738l-.055,2.567a.273.273,0,0,1-.273.267Zm2.8-.239a.288.288,0,0,1-.033-.113l-.226-2.562a5.307,5.307,0,0,1,2.895,1.54l-.005,0-2.256,1.23a.271.271,0,0,1-.137.038A.277.277,0,0,1,8065.145,10962.54Z"
				transform="translate(-7747 -10570)"
				fill="#edede3"
				stroke="rgba(0,0,0,0)"
				stroke-miterlimit="10"
				stroke-width="1"
			></path>
		</g>
	</svg>
</i>

<style>
	svg {
		transform: scale(var(--scale, 1));
	}
</style>
