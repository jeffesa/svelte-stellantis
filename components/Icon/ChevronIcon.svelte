<script>
	export let name = ''
	export let scale = 1
	export let type = 'left'
	export let color = '#fff'
</script>

<i
  class="{$$props.class || ''}"
	class:right="{type === 'right'}"
	class:left="{type === 'left'}"
	class:down="{type === 'down'}"
	class:up="{type === 'up'}"
	style="--scale:{scale}; --color: {color}"
	rel="icon"
	role="{name}"
></i>

<style>
	i {
		color: var(--color, '#fff');
		box-sizing: border-box;
		position: relative;
		display: block;
		transform: scale(var(--scale, 1));
		width: 22px;
		height: 22px;
		border: 2px solid transparent;
		border-radius: 100px;
		&::after {
			content: '';
			display: block;
			box-sizing: border-box;
			position: absolute;
			width: 10px;
			height: 10px;
			border-bottom: 2px solid;
			border-left: 2px solid;
			left: 6px;
			top: 4px;
		}
		&.left::after {
			transform: rotate(45deg);
		}
		&.right::after {
			transform: rotate(220deg);
      left: 2px; 
		}
		&.down::after {
			transform: rotate(-45deg);
		}
		&.up::after {
			transform: rotate(135deg);
		}
	}
</style>
