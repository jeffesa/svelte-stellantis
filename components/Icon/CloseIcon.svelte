<script>
	export let scale = 1.5
  export let color = '#fff'
</script>

<i 
  style="--scale:{scale}; --color: {color}" 
  rel="icon" 
  class="{$$props.class}" 
/>

<style>
	i {
		color: #fff;
		box-sizing: border-box;
		position: relative;
		display: block;
		transform: scale(var(--scale, 1));
		width: 22px;
		height: 22px;
		border: 2px solid transparent;
		border-radius: 40px;
		&::before,
		&::after {
			content: '';
			display: block;
			box-sizing: border-box;
			position: absolute;
			width: 16px;
			height: 2px;
      background: var(--color, ‘#fff’);
			transform: rotate(45deg);
			border-radius: 5px;
			top: 8px;
			left: 1px;
		}
		&::after {
			transform: rotate(-45deg);
		}
	}
</style>
