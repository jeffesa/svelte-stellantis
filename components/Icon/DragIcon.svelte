<script>
	export let scale = 1
</script>

<style>
	svg {
		transform: scale(var(--scale, 1));
	}
</style>

<i rel="icon">
	<svg
		style="--scale:{scale}"
		xmlns="http://www.w3.org/2000/svg"
		width="5.5rem"
		height="5.5rem"
		viewBox="0 0 78.109 75.965"
	>
		<g transform="translate(-1 -1)">
			<ellipse
				cx="37.43"
				cy="37.983"
				fill="#fff"
				rx="37.43"
				ry="37.983"
				transform="translate(1 1)"
			></ellipse>
			<g>
				<path
					fill="#d24f19"
					d="M625.872 978.7l-3.981-3.2c21.118-16.967 21.118-44.573 0-61.54l3.981-3.2c23.314 18.733 23.314 49.21 0 67.94z"
					transform="translate(57.643 4.62) translate(-621.891 -910.764)"
				></path>
			</g>
			<text font-size="0.65rem" font-style="normal" font-weight="600">
        <tspan x="15" y="30">ARRASTE</tspan>
				<tspan x="15" y="43">PARA VER</tspan>
				<tspan x="15" y="56">EM 360ยบ</tspan>
			</text>
		</g>
	</svg>
</i>
