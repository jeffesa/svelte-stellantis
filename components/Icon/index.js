import Icon from './Icon.svelte'
import CloseIcon from './CloseIcon.svelte'
import BrandIcon from './BrandIcon.svelte'
import MenuIcon from './MenuIcon.svelte'
import PlusIcon from './PlusIcon.svelte'
import ChevronIcon from './ChevronIcon.svelte'
import ScrollDownIcon from './ScrollDownIcon.svelte'
import TagIcon from './TagIcon.svelte'
import BrandTypeIcon from './BrandTypeIcon.svelte'
import BuyIcon from './BuyIcon.svelte'
import MountIcon from './MountIcon.svelte'
import SocialIcon from './SocialIcon.svelte'
import DragIcon from './DragIcon.svelte'
import EyeIcon from './EyeIcon.svelte'
import ZoomIcon from './ZoomIcon.svelte'

Icon.Close = CloseIcon
Icon.Brand = BrandIcon
Icon.Menu = MenuIcon
Icon.Plus = PlusIcon
Icon.Chevron = ChevronIcon
Icon.ScrollDown = ScrollDownIcon
Icon.Tag = TagIcon
Icon.BrandType = BrandTypeIcon
Icon.BuyIcon = BuyIcon
Icon.MountIcon = MountIcon
Icon.Social = SocialIcon
Icon.Drag = DragIcon
Icon.Eye = EyeIcon
Icon.Zoom = ZoomIcon

export default Icon

export {
	CloseIcon,
	BrandIcon,
	MenuIcon,
	PlusIcon,
	ScrollDownIcon,
	TagIcon,
	BrandTypeIcon,
	BuyIcon,
	MountIcon,
  SocialIcon,
  DragIcon,
	EyeIcon
}
