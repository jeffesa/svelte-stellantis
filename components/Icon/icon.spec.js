import { render } from '@testing-library/svelte'
import Icon from './Icon.svelte'

test("when don't send the `name` prop render with 'invalid icon name' text", () => {
	const { getByText } = render(Icon)
	expect(getByText('invalid icon name'))
})

test('render icon `brand` by name', () => {
	const role = 'brand'
	const { getByRole } = render(Icon, { name: role })
	expect(getByRole(role))
})

test('render icon `plus` by name', () => {
	const role = 'plus'
	const { getByRole } = render(Icon, { name: role })
	expect(getByRole(role))
})

test('default `plus` icon is `sm`', () => {
	const role = 'plus'
	const { container } = render(Icon, { name: role })
	expect(container).toMatchSnapshot()
	const $icon = container.querySelector('i')
	expect($icon.classList.contains('sm')).toBeTruthy()
})

test('icon `plus` can be `md`', () => {
	const role = 'plus'
	const { container } = render(Icon, {
		name: role,
		md: true
	})
	expect(container).toMatchSnapshot()
	const $icon = container.querySelector('i')
	expect($icon.classList.contains('md')).toBeTruthy()
})

test('icon `plus` can be `lg`', () => {
	const role = 'plus'
	const { container } = render(Icon, {
		name: role,
		lg: true
	})
	expect(container).toMatchSnapshot()
	const $icon = container.querySelector('i')
	expect($icon.classList.contains('lg')).toBeTruthy()
})

test('icon `plus` can be `orange`', () => {
	const role = 'plus'
	const { container } = render(Icon, {
		name: role,
		orange: true
	})
	expect(container).toMatchSnapshot()
	const $icon = container.querySelector('i')
	expect($icon.classList.contains('orange')).toBeTruthy()
})
