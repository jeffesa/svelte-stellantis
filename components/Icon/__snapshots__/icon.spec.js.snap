// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default \`plus\` icon is \`sm\` 1`] = `
<body>
  <div>
    
    
  </div>
</body>
`;

exports[`icon \`plus\` can be \`lg\` 1`] = `
<body>
  <div>
    
    
  </div>
</body>
`;

exports[`icon \`plus\` can be \`md\` 1`] = `
<body>
  <div>
    
    
  </div>
</body>
`;

exports[`icon \`plus\` can be \`orange\` 1`] = `
<body>
  <div>
    
    
  </div>
</body>
`;
